src/components/Chatbox/Input.tsx

```diff
@@ -0,0 +1,241 @@
+
+import { MessageBodyType } from '@graasp/query-client/dist/src/types';
+import { CHATBOX } from '@graasp/translations';
+
+import {
+  charCounterCypress,
+  inputTextFieldCypress,
+  inputTextFieldTextAreaCypress,
+  sendButtonCypress,
+} from '../../config/selectors';
+import {
+  ALL_MEMBERS_DISPLAY,
+  ALL_MEMBERS_ID,
+  HARD_MAX_MESSAGE_LENGTH,
+} from '../../constants';
+import { useCurrentMemberContext } from '../../context/CurrentMemberContext';
+import { useMessagesContext } from '../../context/MessagesContext';
+import { getAllMentions, normalizeMentions } from '../../utils/mentions';
+
+type Props = {
+  id?: string;
+  inputRef: RefObject<HTMLTextAreaElement>;
+  placeholder?: string;
+  textInput: string;
+  setTextInput: (newText: string) => void;
+  sendMessageFunction?: (body: MessageBodyType) => void;
+};
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    width: '100%',
+  },
+  textLength: {
+    whiteSpace: 'pre',
+    paddingLeft: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    color: 'gray',
+  },
+  textTooLong: {
+    color: 'red !important',
+  },
+}));
+
+const inputStyle = {
+  width: '100%',
+  // mentions
+  control: {
+    // fontFamily: 'monospace',
+    minHeight: 63,
+  },
+  input: {
+    padding: 9,
+    border: '1px solid silver',
+    width: '100%',
+    overflow: 'auto',
+    height: 70,
+    borderRadius: '4px',
+  },
+  highlighter: {
+    padding: 9,
+    border: '1px solid transparent',
+    boxSizing: 'border-box',
+    overflow: 'hidden',
+    height: 70,
+  },
+
+  suggestions: {
+    list: {
+      backgroundColor: 'white',
+      border: '1px solid rgba(0,0,0,0.15)',
+      fontSize: 14,
+    },
+    item: {
+      padding: '5px 15px',
+      borderBottom: '1px solid rgba(0,0,0,0.15)',
+      '&focused': {
+        backgroundColor: '#cee4e5',
+      },
+    },
+  },
+};
+
+const mentionStyle = {
+  backgroundColor: '#c5c9e8',
+};
+
+const Input: FC<Props> = ({
+  id,
+  inputRef,
+  textInput,
+  setTextInput,
+  placeholder,
+  sendMessageFunction,
+}) => {
+  const classes = useStyles();
+  const { members } = useMessagesContext();
+  const { id: currentMemberId } = useCurrentMemberContext();
+  const { t } = useTranslation();
+
+  const memberSuggestions = [
+    { id: ALL_MEMBERS_ID, display: ALL_MEMBERS_DISPLAY },
+    ...(members
+      // exclude self from suggestions
+      ?.filter((m) => m.id !== currentMemberId)
+      .map((m) => ({ id: m.id, display: m.name }))
+      .toJS() || []),
+  ] as SuggestionDataItem[];
+  // add mention to all
+  const isMessageTooLong = textInput.length > HARD_MAX_MESSAGE_LENGTH;
+
+  // autofocus on first render
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, []);
+
+  const onSend = (): void => {
+    if (textInput) {
+      const mentions = getAllMentions(textInput).map(({ id }) => id);
+      // expand '@all' to all members in mentions array
+      let expandedMentions = mentions;
```

Comment:
Do the following and remove the casting at line 138.
```suggestion
let expandedMentions: string[] = mentions;
```